Formula/hcl2json.rb

```diff
@@ -0,0 +1,33 @@
+class Hcl2json < Formula
+  desc "Convert hcl2 to json"
+  homepage "https://github.com/tmccombs/hcl2json"
+  url "https://github.com/tmccombs/hcl2json/archive/v0.3.3.tar.gz"
+  sha256 "e2aa5ef900cfe42ebd9454cfe61b8cf780b4a026dae22e4ef5fc779f34da4126"
+  license "Apache-2.0"
+  head "https://github.com/tmccombs/hcl2json.git"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args
+  end
+
+  test do
+    require "json"
+    require "open3"
+    test_plaintext = "This is plain text."
+    test_hcl = <<~HCL
+      resource "my_resource_type" "test_resource" {
+        input = "magic_test_value"
+      }
+    HCL
+    hcl_stdout, hcl_status = Open3.capture2(bin/"hcl2json", stdin_data: test_hcl)
+    assert_equal 0, hcl_status, "hcl2json gave the wrong exit code"
+    hcl_parsed = JSON.parse(hcl_stdout)
+    assert_equal "magic_test_value",
+                 hcl_parsed["resource"]["my_resource_type"]["test_resource"][0]["input"],
+                 "Parsed HCL returned incorrect data"
+    _plaintext_stdout, plaintext_status = Open3.capture2e(bin/"hcl2json", stdin_data: test_plaintext)
+    assert_equal 256, plaintext_status, "hcl2json should exit 256 on plaintext input"
```

Comment:
```suggestion
    test_hcl = <<~HCL
      resource "my_resource_type" "test_resource" {
        input = "magic_test_value"
      }
    HCL

    test_json = {
      resource: {
        my_resource_type: {
          test_resource: [
            {
              input: "magic_test_value"
            }
          ]
        }
      }
    }.to_json

    assert_equal test_json, pipe_output("#{bin}/hcl2json", test_hcl).gsub(/\s+/, "")
```
And you could similarly add the plaintext check if you wish to.

Reply:
Ahhh this isn't my usual JSON testing style but I can definitely see how it makes the comparison a _lot_ clearer to a casual reader. 

Give me a few minutes to update the test and I'll revise the PR. Thanks!

Reply:
@nandahkrishna I revised the technique for the HCL to JSON test as suggested

However, my attempts to use `pipe_output` to test the plaintext input return code behavior all resulted in the test barfing `hcl2json`'s stderr text on the screen in a manner suggestive of an error even when the test was good, like so:
![stderr barf](https://user-images.githubusercontent.com/539074/125532845-c3d7fe19-5d7e-400b-8f2b-af840a5b91ec.png)

I don't like this, but compared to what I've submitted here, the _code_ is less verbose:
```ruby
pipe_output("#{bin}/hcl2json", test_plaintext, 1)
```

I've noticed that Homebrew tends to prefer shorter formulae though. Would you like me to switch the test to `pipe_output`?

Reply:
This might solve the issue you're talking about:
```
pipe_output("#{bin}/hcl2json 2>&1", test_plaintext, 1)
```
We could definitely do away with unnecessary variables and keep the `test` block concise. After fixing style errors, it should probably look like this:
```
  test do
    test_hcl = <<~HCL
      resource "my_resource_type" "test_resource" {
        input = "magic_test_value"
      }
    HCL

    test_json = {
      resource: {
        my_resource_type: {
          test_resource: [
            {
              input: "magic_test_value",
            },
          ],
        },
      },
    }.to_json

    assert_equal test_json, pipe_output("#{bin}/hcl2json", test_hcl).gsub(/\s+/, "")
    assert_match "Failed to convert", pipe_output("#{bin}/hcl2json 2>&1", "Hello, Homebrew!", 1)
  end
```